### PR TITLE
Accessibility: Screen readable breadcrumbs and improved dropdown

### PR DESCRIPTION
**Pull request by pxwxnvermx**
## Product Description
This PR includes changes for improving the order in which breadcrumbs on webapp menus are read and improves the accessibility of the dropdown.

## Technical Summary
[Ticket Link](https://dimagi-dev.atlassian.net/browse/ush-3044)

### Safety story

Tested locally and on staging.

### QA Plan
[QA Ticket](https://dimagi-dev.atlassian.net/browse/QA-5138)

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js [hunk 1]
```diff
@@ -120,6 +120,11 @@ hqDefine("cloudcare/js/formplayer/menus/controller", function () {
             menusUtils.showBreadcrumbs(menuResponse.breadcrumbs);
             if (!appPreview && ((menuResponse.langs && menuResponse.langs.length > 1 && changeFormLanguage) || enablePrintOption)) {
                 menusUtils.showFormMenu(menuResponse.langs, changeFormLanguage);
+                // Make menu the last element in the breadcrumbs
+                const formMenuEl = $("#form-menu");
+                const formMenuElParent = formMenuEl.parent();
+                formMenuEl.remove();
+                formMenuElParent.append(formMenuEl);
```
Comment (mkangia) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js:127]: I am confused for why we are doing this here and not in html itself?

File: corehq/apps/cloudcare/templates/formplayer/menu_list.html [hunk 2]
```diff
@@ -128,9 +128,17 @@ <h1 class="page-title"><%- title %></h1>
...
-     <ul class="dropdown-menu dropdown-menu-right noprint-sub-container">
+  <div id="form-menu-dropdown">
+    <a class="dropdown-toggle"
+       title="{% trans 'Dropdown toggle' %}"
+       role="button"
+       href="javascript:void(0);"
+       data-toggle="dropdown"
+       aria-haspopup="true"
+       aria-expanded="false">
+      <i class="fa fa-bars" style="color: #fff"></i>
```
Comment (mkangia) [inline corehq/apps/cloudcare/templates/formplayer/menu_list.html:139]: Can the styling be in stylesheet?

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js (hunk 1 above)
Comment (pxwxnvermx) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js:127]: The element for the dropdown is already present in the html, breadcrumbs are dynamically added by the javascript. That causes the dropdown to be the second item in the list and not the last one which it should be. The code above removes the element from the 2nd spot and adds it to the end of the list.

Comment (mkangia) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js:127]: > The code above removes the element from the 2nd spot and adds it to the end of the list.

Relying on the order seems risky. Can we have a more reliable way?

> breadcrumbs are dynamically added by the javascript.

May be at this point we could ensure the order we need?

File: corehq/apps/cloudcare/templates/formplayer/menu_list.html (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/cloudcare/templates/formplayer/menu_list.html:139]: I don't think this markup is right. The font color styling caught my eye, it's rare to need to set font color in our codebase. Looking at the [dropdown docs](https://getbootstrap.com/docs/3.3/components/#btn-dropdowns), my guess is that the `a` should be a button and the `form-menu-dropdown` div may need to be a `btn-group`.

Comment (pxwxnvermx) [reply corehq/apps/cloudcare/templates/formplayer/menu_list.html]: Fixed. [754b612](https://github.com/dimagi/commcare-hq/pull/32995/commits/754b61277d0663d7ffcc354990922269931cd251)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js (hunk 1 above)
Comment (pxwxnvermx) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js]: Fixed. [e097bce](https://github.com/dimagi/commcare-hq/pull/32995/commits/e097bce5650d937780b841d8ced6fc2264d8f258)

image

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 3]
```diff
@@ -709,7 +709,6 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
 
     const FormMenuView = Marionette.CollectionView.extend({
         template: _.template($("#form-menu-template").html() || ""),
-        tagName: 'li',
```
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js:712]: why does this need to be removed?

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js]: Thank you! I didn't review the new code closely, but I appreciate the change in direction, and from skimming it, it looks simpler.

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js (hunk 3 above)
Comment (pxwxnvermx) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js:712]: The dropdown was getting rendered under a <li> tag and that caused a bullet icon to appear in front of the dropdown. Removing this line makes the dropdown render under a <div> tag.

**Comment by mkangia** [comment]
thanks for the follow up @pxwxnvermx 
Will defer to @orangejenny or @stephherbers for approval on this

**Review by stephherbers — APPROVED** [review]

**Pull request merged**